### ISSUE TITLE
restrict-plus-operands should not flag operands of type `any`

### ISSUE THREAD

**Issue by johnsoft** [palantir/tslint#3954]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const x = 'x';
const y: any = 'y';
const z = x + y;
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "restrict-plus-operands": true
  }
}
```

#### Actual behavior

```
ERROR: 3:11  restrict-plus-operands  Operands of '+' operation must either be both strings or both numbers, consider using template literals
```

#### Expected behavior

I don't expect an error here. Generally when you're using `any` it's because you want to be as permissive as possible. In my case the `any` comes from an external module and casting the value to a string would just be noise.

**Comment by Kani1013** [comment]
I think this error is expected and useful here.
Try using string literals, that should work:

````typescript
const x = 'x';
const y: any = 'y';
const z = `${x}${y}`;
````

**Comment by giladgray** [comment]
I agree with @Kani1013. this rule is intended to be more strict than `tsc` (which does allow addition on `any`).
